src/plugins/vctlGit.nim

```diff
@@ -412,6 +415,18 @@ proc loadObject(info: RepoInfo, refId: string): Table[string, string] =
        objData.endsWith(gpgSignEnd):
       result[gitSign] = ""
 
+    # Set the git commit message.
+    # If the commit is signed, the commit message appears after the
+    # end of the signature.
+    result[gitCommitMessage] = block:
+      let iGpgSignEnd = objData.find(gpgSignEnd)
+      let iMessageStart =
+        if iGpgSignEnd != -1:
+          iGpgSignEnd + gpgSignEnd.len
+        else:
+          objData.find("\n\n")
+      objData[iMessageStart ..< objData.len].strip()
```

Comment:
there are different git objects types in git. both of them them have common structure which is it has headers and then the object body.

for git commits it has a header field `gpgsig` which contains the signature and then the body of the object file is the commit message. it gets slightly tricky as the `gpgsign` field has `\n\n` so that needs to be accounted for.

```
➜ git cat-file -t 54d6df1dc2782ed0bd5eb036569f3224f0eb4cdc
commit

➜ git cat-file commit 54d6df1dc2782ed0bd5eb036569f3224f0eb4cdc
tree 4d5fcadc293a348e88f777dc0920f11e7d71441c
author Miroslav Shubernetskiy <miroslav@miki725.com> 1708697830 -0500
committer Miroslav Shubernetskiy <miroslav@miki725.com> 1708697830 -0500
gpgsig -----BEGIN PGP SIGNATURE-----

 iQIzBAABCgAdFiEE6X2pNQoznpT+LzPmmRF166MKTa0FAmXYqOoACgkQmRF166MK
 Ta0VYA/9ELUzX3vsXWBS3n4WA/iEKwVTtWJCdTiV0/wJntlUvhbC4PCaygEWFtKM
 KLRx7pIwjcGO6LKVirEOOpGX8o4J/rIukvRq94NH0WfHFMBOVfzVk9/PlOz9y+Sp
 SH9RhOiwkP5O1fv1ds00RZf7Hns7vJWeEUuAFksRzNPzhAYRWPZoxYUh8SgXwK1D
 pYClQ/krFBtP33jFjWe/JqUQ4q9hVUek5ruVk+kBkFtx4Ivb4fkfH7XvH7MbJ1dm
 Zi67Wet4yAixH9vepRpbrYpy+CQWwjbCrUS2piqrs7Z2JAogv401OTdVRzZLUpeR
 CV9yQ2ohC6ay7pMZo6T503K4VZBva5p8DWTg6ni2bWrhmzGBJiiBCpRwQxPC90is
 RRQkDlTqeb49h4K1HbRD3gycgrgLwaiqoZCt+p6e8TD3iG/VdBZqv+D+66TP+gEN
 Vj2+uapenvZVd/Uw2ml2a9QAj4hJt+ZF2js5nY1FJRN5WUeOfNm0KIPBWvCq4qoy
 VvHv8aWFyJ9rfJEZDfhvAZbNWGG8gAKDFVGMIalQoBFjwWdAxyde1lGK0l+Ws5gi
 8Kczq4mw6mn4sSGK2VGX1esJX8kRasz0Gf1Pp+tt5U7FsYl2d0SNiEzFNjrKB29k
 hNOn+KOwksr3x7i27pUkkMuEmkvLSF8dGymK77j6EpS+wu3EK3g=
 =jfHe
 -----END PGP SIGNATURE-----

initial
```

for git tag objects (which only exist for annotated tags), thats different though as signature is at the end of the message itself:

```
➜ git cat-file -t 506992de14a7cf63db9cd81480ba94f25c38032b
tag

➜ git cat-file tag 506992de14a7cf63db9cd81480ba94f25c38032b
object 54d6df1dc2782ed0bd5eb036569f3224f0eb4cdc
type commit
tag v1
tagger Miroslav Shubernetskiy <miroslav@miki725.com> 1708697867 -0500

tag commit
-----BEGIN PGP SIGNATURE-----

iQIzBAABCgAdFiEE6X2pNQoznpT+LzPmmRF166MKTa0FAmXYqQsACgkQmRF166MK
Ta2R+w/+Oqx7TsV0Opm+LSLCxzQDBJpLSEmTdBPKkKhTC//lltP0urygryeL/j8G
1IwyIaPHXjLqFwUKjfDbvSipZWwAMO9Kdd+PB/lb7RO3LgLl3djswEeQWnol+937
hy49OnGDsRmZ5jA/eoxLlfMITZHSQoQ9MD+kNS6D6W4gkChm2WB4aPkP4/sOblKg
4eaZUI/f+XsLuWdfSpogDT5uivJYp4P9kVTRkz01S7G3BUDzfdpXG8YsIkOEhNNV
ADP+lCln0HizUDxk55uTQH+sHfpg9OdJ2Mk/0yOz8mQSc34IVpHErFZk84Q1yhsv
T94tjOC/0XGspKH/gesqxcgO4cpZVH4ON9SAtgJDRachape74YV9VKhHJYYLN31F
tL26brC6SBcPE9MrQ/lRjCxpxXsKP+IDwK62RMMfbSgkucugYwNOi6aKCMfF94ww
uvEh3WTTz3Mkd9oQJfA1HUEHPqGZjsrWIqYQJzMeWHJ3R8dwYo2FtWAcjH8sIbxI
7lLjjU8TnNc3YpUs9CfDi+ax9HcLLz1q/+cKfTgobrmw34bU+LKvrh57ycrG0aKc
h4KWMttpzvXmP7maWs97CiRTxKA5eM9vM5vz9ftwSDvC4TVgRV3JJ3qKcFRDfNqG
BAB348mOFHy548Z6AerJCf+Cj9dpsGthkIOJjrY3AxzmOvRAV/o=
=QybJ
-----END PGP SIGNATURE-----
```

this is inside `loadObject` function which is used to parse both commit/tag object files so the implementation should work for both